src/refactor/index.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+
+
+// return on each Click the value of Square
+function Square(props) {
+  return (
+    <button className="square" onClick={props.onClick}>
+      {props.value}
+    </button>
+  )
+}
+// Describe Board Game + set boolean to setup user 
+class Board extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      squares: Array(9).fill(null),
+      xIsNext: true,
+    };
+  }
+
+  // handle which turn is it
+  handleClick(i) {
+    const squares = this.state.squares.slice();
+    squares[i] = this.state.xIsNext ? 'X': 'O';
+    this.setState({
+      squares: squares,
+      xIsNext : !this.state.xIsNext,
+    });
+    
+}
+
+
+  renderSquare(i) {
+    return (
+      <Square
+        value={this.state.squares[i]}
+        onClick={() => this.handleClick(i)}
+      />
+    );
+    
+   
+  }
+  
+  render() {
+    
+    const winner = calculateWinner(this.state.squares);
+    let status;
+    if (winner) {
+      status = 'Winner:' + winner
+    } else {
+      status = 'Next player:' + (this.state.xIsNext ? 'X' : 'O');
+    }
+
+    return (
+      <div>
+        <div className="status">{status}</div>
+        <div className="board-row">
+          {this.renderSquare(0)}{this.renderSquare(1)}{this.renderSquare(2)}
```

Comment:
Why do you use a function to wrap a call for a component?

Reply:
I wanted at render to decided who wins  +  change the status dynamically, there is a better way?

Reply:
Yes, I would do something like that:
1. initially I would bind the click handler so I can reuse it for all squares and also use a more meaningful naming:
```
const markingASquare = this.clickHandler.bind(this)
```
2. I would change a bit the clickHandler interface with a concept called partial application so it will be prettier later to pass this method as `onClick` argument (I simply wrapped the method logic with additional returned arrow function):
```
handleClick(i) {
    return () => {
        const squares = this.state.squares.slice();
        squares[i] = this.state.xIsNext ? 'X': 'O';
        this.setState({
          squares: squares,
          xIsNext : !this.state.xIsNext,
        });
    }
}
```

3. I would get rid of `renderSquare` method and simply call the squares:
```
<Square
  value={this.state.squares[0]}
  onClick={markingASquare(0)}
/>
<Square
  value={this.state.squares[1]}
  onClick={markingASquare(1)}
/>
<Square
  value={this.state.squares[2]}
  onClick={markingASquare(2)}
/>
```